pom.xml

```diff
@@ -380,7 +400,7 @@
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
-               <version>3.14.0</version>
+               <version>3.15.0</version>
```

Comment:
This causes a bunch of errors. I think this has to be left at 3.14.0 until we upgrade to Java 8+.

Reply:
will look more at this one.

Reply:
Issue is with ASM.  This may be related to the fact that maven is now suggesting all plugins that use maven core components use provided scope.  In the plugin section, if maven-pmd-plugin is added with asm 9.2, the warnings go away.  Maven pmd plugin is using maven-core 3.1.0 in compile scope with 3.15.0.  It used maven-embedder in provided scope in 3.14.0.  Both versions pick up asm 3.3.1 so based on my testing, 3.3.1 is winning resolution order.  The group id is different between 3.3.1 and 9.2.  This can be shown more directly by using 3.3.1 as it produces same output in that case.  I'm pushing up a small patch here but right place to fix is with pmd.  I had not noticed in my other usage as I typically override pmd to latest which also solves the same issue.

Reply:
pmd is fixed on master, they moved to maven core 3.1.1 and it doesn't contain asm.  I'm going to submit a patch to them anyways since newer maven is issuing warnings on plugins to not use compile scope.